provider/lxd/credentials.go

```diff
@@ -172,6 +173,24 @@ func (p environProviderCredentials) DetectCredentials() (*cloud.CloudCredential,
 	}, nil
 }
 
+// DetectCredentialsFor is part of the environs.ProviderCredentialsCloudSpecific interface.
+func (p environProviderCredentials) DetectCredentialsFor(cloudName string) (*cloud.CloudCredential, error) {
+	creds, err := p.detectRemoteCredentials()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	cred, ok := creds[cloudName]
+	if !ok {
+		return nil, errors.NotFoundf("credential for %s", cloudName)
+	}
+	return &cloud.CloudCredential{
+		DefaultCredential: cloudName,
```

Comment:
Since there's just one we don't really need to set DefaultCredential but doesn't hurt to do so